okhttp-tests/src/test/java/okhttp3/internal/framed/HttpOverSpdyTest.java

```diff
@@ -582,6 +583,99 @@ protected HttpOverSpdyTest(Protocol protocol) {
     assertEquals(0, server.takeRequest().getSequenceNumber());
   }
 
+  @Test public void recoverFromOneRefusedStreamReusesConnection() throws Exception {
+    server.enqueue(new MockResponse()
+        .setSocketPolicy(SocketPolicy.RESET_STREAM_AT_START)
+        .setHttp2ErrorCode(ErrorCode.REFUSED_STREAM.httpCode));
+    server.enqueue(new MockResponse()
+        .setBody("abc"));
+
+    Call call = client.newCall(new Request.Builder()
```

Comment:
You saw something I didn’t see! I turned that on and of course the test started to fail. I fixed it by changing `hasMoreRoutes()` to return true if the current route is still valid.